packages/playwright-core/src/server/utils/network.ts

```diff
@@ -64,7 +64,7 @@ export function httpRequest(params: HTTPRequestParams, onResponse: (r: http.Inco
     } else {
       (parsedProxyURL as any).secureProxy = parsedProxyURL.protocol === 'https:';
 
-      options.agent = new HttpsProxyAgent(parsedProxyURL);
+      options.agent = new HttpsProxyAgent(url.format(parsedProxyURL));
```

Comment:
- I'm not sure `url.format()` understands `secureProxy` option.
- Shall we switch to `new URL(proxyURL)` instead of `parse()` + `format()`?

Reply:
Which is why I don't understand why we had this in the first place.

Reply:
Ahh, this is where it was read! https://github.com/TooTallNate/node-http-proxy-agent/blob/5368fa899134a9b0d7d49cc6917c1eecabe7aeb0/src/agent.ts#L54